lib/infra/infra-stack.ts

```diff
@@ -79,35 +82,71 @@ export class InfraStack extends Stack {
     const ec2InstanceType = (props.cpuType === AmazonLinuxCpuType.X86_64)
       ? InstanceType.of(InstanceClass.C5, InstanceSize.XLARGE) : InstanceType.of(InstanceClass.C6G, InstanceSize.XLARGE);
 
+    const alb = new NetworkLoadBalancer(this, 'publicNlb', {
+      vpc: props.vpc,
+      internetFacing: true,
+      crossZoneEnabled: true,
+    });
+
+    if (!props.securityDisabled && !props.minDistribution) {
+      opensearchListener = alb.addListener('opensearch', {
+        port: 443,
+        protocol: Protocol.TCP,
+      });
+    } else {
+      opensearchListener = alb.addListener('opensearch', {
```

Comment:
Did you try to deploy this set up? CDK will throw an error for constructs with same logical resource id. `opensearch` in this case

Reply:
Yes, have deployed multiple clusters with the current change. the listener is tied to the load balancer so as long as load balancer is different the listeners having same identifier wouldn't break it.